testRequirement.coffee

```diff
@@ -62,35 +70,82 @@ matchesSubstring = (objectKey, substringTest, object) ->
         re = RegExp substring, "im" # case insensitive, multiline matching
       else
         re = RegExp substring, "m" # multiline matching
-      if objectValue
-        objectValue = objectValue.match re
+      if objectValueStr
+        match = re.test objectValueStr
+
+      # if substring in object value str
+      return true if match
+  return false
+
+# Check if the test is an empty test
+isEmptyTest = (testValues) ->
+  if (_.isPlainObject testValues) and 'empty' of testValues then true else false
+
+# Check if specific field value is empty
+isEmpty = (objectKey, emptyTest, object) ->
+  if objectKey not of object
+    # this field does not exist so return true that it's "empty"
+    # TODO: throw an error
+    return true
+
+  objectValue = object[objectKey]
+  objectValueIsCorrectType = (objectValue isnt null) and (typeof objectValue isnt 'boolean')
+  shouldBeEmpty = emptyTest['empty']
+  if typeof shouldBeEmpty isnt 'boolean'
+    # TODO: throw an error
+    return true
+
+  if shouldBeEmpty and objectValueIsCorrectType #return true if objectValue is empty
+    return objectValue.length is 0
+
+  else if not shouldBeEmpty and objectValueIsCorrectType #return true if objectValue is not empty
+    return objectValue.length isnt 0
+
+  else #objectValue is not correct type, so makes no sense to check if it's empty
+    # TODO: throw an error
+    return true
 
-      # if substring in object value
-      return true if objectValue
+
+
+# Check if the test is a pubdate test
+isPubdateTest = (testValues) ->
+  if (_.isPlainObject testValues) and 'year' of testValues then true else false
+
+# Extract specific time from the document pubdate field and compare it with the test
+matchesPubdate = (objectKey, pubdateTest, object) ->
+  objectValue = object[objectKey]
+  year = new Date(objectValue).getFullYear() # convert to milliseconds
+
+  if 'year' of pubdateTest # compare years
+    testValue = pubdateTest['year']
+
+    if testValue is 'now' # compare pubdate year value with year now
+      thisYear = new Date().getFullYear()
+      return year is thisYear
+
+  # TODO expand test to month, etc
   return false
 
 # Tests if testValue is in object, either directly or in array
 testValueInObject = (objectKey, testValue, object) ->
+  if objectKey not of object
+    return false
   objectValue = object[objectKey]
-  if objectKey is 'originurl'
-    request.get({url: objectValue, followRedirect: false}, (err, response) =>
-      objectValue = response.headers.location
-      return testValue is objectValue
-    )
-  else
-    if util.isArray objectValue
-      return (objectValue.indexOf testValue) isnt -1
-    else # objectValue is not an array, compare directly
-      return testValue is objectValue
+  objectValue = JSON.stringify(objectValue) # match in stringified arrays and objects too
+  re = RegExp testValue, "im"
+  match = re.test objectValue
```

Comment:
just do `return re.test objectValue` no need to store match and return it right after